### PR TITLE
Refactor deisctl config logic and add docs on Router SSL

### LINKED ISSUE #2712: deisctl config router set sslCert="$(cat cert.crt)" not working
deisctl 1.0.2
deis 1.0.2
deis cluster running on 3 digitalocean instances (4GB) 

When attempting to set the router sslCert and sslKey, I get the following:

$ deisctl config router set sslCert="$(cat cert.crt)"
Error: invalid argument: sslCert=-----BEGIN CERTIFICATE-----
......
......
......
-----END CERTIFICATE-----

So I tried to get into deis-1 and set it directly via etcdctl:

$ etcdctl set /deis/router/sslCert $(cat cert.crt)
Incorrect Usage.

NAME:
   set - set the value of a key

USAGE:
   command set [command options] [arguments...]

DESCRIPTION:

OPTIONS:
   --ttl '0'                    key time-to-live
   --swap-with-value            previous value
   --swap-with-index '0'        previous index

I've made sure the cert.crt exists in the directory I am running the commands above from and echo $(cat cert.crt) returns the certificate.

### PR DESCRIPTION
**Pull request by gabrtv**
This PR fixes #2712 and cleans up the config logic in `deisctl`:

 * defines `fileKeys` which are read from local files
 * defines `b64Keys` which are base64-encoded before stored (needed for `sshPrivateKey`)
 * `valueForPath` is factored out with corresponding test coverage

I've also added documentation on how to configure the Deis routers to terminate SSL.

### CODE REVIEW

**Comment by mboersma** [comment]
Code LGTM.

File: deisctl/config/config.go [hunk 1]
```diff
@@ -106,13 +104,31 @@ func doConfigGet(client *etcdClient, root string, keys []string) ([]string, erro
...
+		if path == p {
+
+			// read value from filesystem
+			bytes, err := ioutil.ReadFile(utils.ResolvePath(v))
+			if err != nil {
+				return "", err
+			}
+
+			// see if we should return base64 encoded value
+			for _, pp := range b64Keys {
```
Comment (carmstrong) [inline deisctl/config/config.go]: So b64keys can only be used if they are also fileKeys, and only makes sense in that context? I guess that makes sense.

**Comment by carmstrong** [comment]
Code LGTM.

File: deisctl/config/config.go (hunk 1 above)
Comment (gabrtv) [inline deisctl/config/config.go]: Yep.  The base64 encoding is to sanitize multi-line input, which is typically coming from a file.

**Pull request merged**